### ISSUE TITLE
renderer/Scene: Paint transformation after scene change

### ISSUE THREAD

**Issue by lpogic** [thorvg/thorvg#2958]
If paint is rendered in one scene and then we want to move it to another, its transform will not be updated:

```CPP
auto scene1 = tvg::Scene::gen();
auto scene2 = tvg::Scene::gen();
scene2->translate(100, 100);

auto shape = tvg::Shape::gen();
shape->appendRect(0, 0, 100, 100, 10, 10);
shape->fill(0, 255, 0);

scene1->push(shape);
canvas->push(scene1);
canvas->push(scene2);
canvas->update();

scene1->clear(false);
scene2->push(shape);
canvas->update();
```

Output:
![2](https://github.com/user-attachments/assets/b01e6c7a-a950-4be2-a184-0322b666c032)

Expected output:
![1](https://github.com/user-attachments/assets/2e31bc86-2775-49ed-8571-5489615624f6)

**Comment by hermet** [comment]
@lpogic Hello, besides this issue handling, I would like to recommend using ThorVG in a sequence like this for your optimal performance.

```
1.  paint preparation //paint create & set properties & organize scene tree
2.  canvas->push()  //update
3.  canvas->draw()
```

This is current your usage:
```
1. paint preparation  //paint create & set properties
2. canvas->push() //update
3. paint preparation again //reorganize the scene
4. canvas->update()  //update gain
5. canvas->draw()
```


Recommended code:

```
auto scene1 = tvg::Scene::gen();
auto scene2 = tvg::Scene::gen();
scene2->translate(100, 100);

auto shape = tvg::Shape::gen();
shape->appendRect(0, 0, 100, 100, 10, 10);
shape->fill(0, 255, 0);

scene1->push(shape);

scene1->clear(false);
scene2->push(shape);

canvas->push(scene1);
canvas->push(scene2);
```

**Comment by lpogic** [comment]
@hermet Thank you for suggestion. In my case I am working in an infinite loop:
```
loop {
  # create/modify some paints
  # update nessesary paint tree parts
}
```
The error occurs when I try to transfer paint that has been previously updated between scenes.

In the example, I wanted to more clearly illustrate the bug.

**Comment by hermet** [comment]
@lpogic the fix will be in v0.15.5 thanks.
